docs/ONNX_Runtime_Mobile_NNAPI_perf_considerations.md

```diff
@@ -0,0 +1,111 @@
+# ONNX Runtime Mobile: Performance Considerations When Using NNAPI
+
+ONNX Runtime Mobile with the NNAPI Execution Provider (EP) can be used to execute ORT format models on Android platforms using NNAPI. This document explains the details of how different optimizations affect performance, and provides some suggestions for performance testing with ORT format models. 
+
+Please review the introductory details for [using NNAPI with ONNX Runtime Mobile](ONNX_Runtime_for_Mobile_Platforms.md#Using-NNAPI-with-ONNX-Runtime-Mobile) first.
+
+
+## 1. ONNX Model Optimization Example
+
+ONNX Runtime applies optimizations to the ONNX model to improve inferencing performance. These optimizations occur prior to exporting an ORT format model. See the [graph optimization](ONNX_Runtime_Graph_Optimizations.md) documentation for further details of the available optimizations.
+
+It is important to understand how the different optimization levels affect the nodes in the model, as this will determine how much of the model can be executed using NNAPI.
+
+*Basic*
+
+The _basic_ optimizations remove redundant nodes and perform constant folding. Only ONNX operators are used by these optimizations when modifying the model.
+
+*Extended*
+
+The _extended_ optimizations replace one or more standard ONNX operators with custom internal ONNX Runtime operators to boost performance. Each optimization has a list of EPs that it is valid for. It will only replace nodes that are assigned to that EP, and the replacement node will be executed using the same EP.
+
+*Layout*
```

Comment:
Do you want to add some note that this one is "--optimization_level all" in the script?

Reply:
Not at this point. The script complains noisily if they try and use it, and if I don't mention it they may be less likely to try and use it. :)